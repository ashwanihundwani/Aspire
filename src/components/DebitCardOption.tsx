import React from 'react'
import { View, StyleSheet, Switch, Image, Text, TouchableOpacity } from 'react-native';
import { layouting, AppColors, CommonString } from '../utils/constants';
import { windowWidth, avenirFont, FontStyle, windowHeight } from '../utils/utils';

const spacing = layouting.spacingFactor
const width = windowWidth() - (spacing * 2)
const textSpacingMarginRight = 100
const textWidth = width - textSpacingMarginRight

export interface CardOptionProps {
    title: string,
    description: string,
    icon: any,
    showSwitch: boolean,
    switchEnabled?: boolean,
}

export interface CardOptionDisplayProps extends CardOptionProps {
    id: string,
    index: number
    onSwitchValueChanged?: (on: boolean, index: number) => void
    onClick: (index: number) => void
}
const DebitCardOption: React.FC<CardOptionDisplayProps> = (props: CardOptionDisplayProps) => {
    return (
        <TouchableOpacity
            testID={props.id}
            onPress={() => {
                // For this exercise, handling only Weekly spend action.
                if (props.index === 1) {
                    props.onClick(props.index)
                }
            }}
            style={styles.container}>
            <View style={styles.content}>
                <Image source={props.icon} />
                <View style={styles.textContainer}>
                    <Text style={styles.title} >{props.title}</Text>
                    <Text style={styles.description}>{props.description}</Text>
                </View>
            </View>
            {props.showSwitch ? <Switch testID={props.id + CommonString.switchSuffix} onValueChange={(value) => {
                props && props.onSwitchValueChanged && props.onSwitchValueChanged(value, props.index)
            }} value={props.switchEnabled} /> : null}
        </TouchableOpacity>
    )
}

export default React.memo<CardOptionDisplayProps>(DebitCardOption);

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        justifyContent: "space-between",
        marginBottom: 34,
        width: width,
        marginHorizontal: spacing
    },
    content: {
        flexDirection: "row",
        justifyContent: "flex-start",
    },
    textContainer: {
        marginLeft: 12,
    },
    title: {
        color: AppColors.debitCardTitleColor,
        fontFamily: avenirFont(FontStyle.Medium),
        fontSize: 14,
        width: textWidth,
        marginTop: 4
    },
    description: {
        marginTop: 2,
        color: AppColors.darkTextColor,
        fontFamily: avenirFont(FontStyle.Regular),
        fontSize: 13,
        width: textWidth,

    }
})