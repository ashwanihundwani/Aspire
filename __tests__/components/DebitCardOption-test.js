/**
 * @format
 */

import 'react-native';
import React from 'react';
import DebitCardOption from '../../src/components/DebitCardOption';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
      .create(<DebitCardOption 
                title={"Test"} 
                index={0} 
                description={"Test Description"} 
                onClick={()=> {}}
                showSwitch={true}
                icon={{}} 
                id={"1"}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });