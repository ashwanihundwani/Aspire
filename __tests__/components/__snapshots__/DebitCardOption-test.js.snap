// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 34,
      "marginHorizontal": 43.5,
      "opacity": 1,
      "width": 663,
    }
  }
  testID="1"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      source={Object {}}
    />
    <View
      style={
        Object {
          "marginLeft": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#25345F",
            "fontFamily": "AvenirNext-Medium",
            "fontSize": 14,
            "marginTop": 4,
            "width": 563,
          }
        }
      >
        Test
      </Text>
      <Text
        style={
          Object {
            "color": "#22222266",
            "fontFamily": "AvenirNext-Regular",
            "fontSize": 13,
            "marginTop": 2,
            "width": 563,
          }
        }
      >
        Test Description
      </Text>
    </View>
  </View>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    testID="1_Switch"
    value={false}
  />
</View>
`;
